docs/containers-registries.conf.5.md

```diff
@@ -18,42 +18,49 @@ VERSION 2 is the latest format of the `registries.conf` and is currently in
 beta. This means in general VERSION 1 should be used in production environments
 for now.
 
-Every registry can have its own mirrors configured.  The mirrors will be tested
-in order for the availability of the remote manifest.  This happens currently
-only during an image pull.  If the manifest is not reachable due to connectivity
-issues or the unavailability of the remote manifest, then the next mirror will
-be tested instead.  If no mirror is configured or contains the manifest to be
-pulled, then the initially provided reference will be used as fallback.  It is
-possible to set the `insecure` option per mirror, too.
-
-Furthermore it is possible to specify a `prefix` for a registry.  The `prefix`
-is used to find the relevant target registry from where the image has to be
-pulled.  During the test for the availability of the image, the prefixed
-location will be rewritten to the correct remote location.  This applies to
-mirrors as well as the fallback `location`.  If no prefix is specified, it
-defaults to the specified `location`.  For example, if
-`prefix = "example.com/foo"`, `location = "example.com"` and the image will be
-pulled from `example.com/foo/image`, then the resulting pull will be effectively
-point to `example.com/image`.
-
-By default container runtimes use TLS when retrieving images from a registry.
-If the registry is not setup with TLS, then the container runtime will fail to
-pull images from the registry. If you set `insecure = true` for a registry or a
-mirror you overwrite the `insecure` flag for that specific entry.  This means
-that the container runtime will attempt use unencrypted HTTP to pull the image.
-It also allows you to pull from a registry with self-signed certificates.
-
-If you set the `unqualified-search = true` for the registry, then it is possible
-to omit the registry hostname when pulling images.  This feature does not work
-together with a specified `prefix`.
+The configuration is a sequence of `[[registry]]` entries.  Given an input image
+name, a single entry is chosen; that entry must have a `prefix` field set to be
+a prefix of the input image name (taking into account namespace/repo/tag/digest
+separators), and (only) the entry with the longest matching `prefix` field
+is used.
+
+As a special case, the `prefix` field can be missing; if so, it defaults to the value
```

Comment:
For me this should be the default case, because the feature is way more easy to understand when initially omitting the `prefix` semantics.

Reply:
I thought it makes more sense to focus on `prefix` rather than `location`, because _everyone_ editing the configuration needs to understand the semantics of selecting the entry, and that it is `prefix`-based, whereas actual remapping using a `prefix`/`location` pair is only useful for, presumably, much  rarer cases of sites disconnected from the internet.

So, we can have the discussion of `location` and mirrors in a separate section relevant only for people who care, and allow, notably, the `insecure` flag to usable without having to understand `location` at all.